src/components/CollapsibleV2/Collapsible.tsx

```diff
@@ -0,0 +1,199 @@
+import React, { ReactNode, useState, useRef, useId, useEffect } from 'react'
+import styled from '@emotion/styled'
+import {
+  color,
+  space,
+  fontWeight,
+  radius,
+  device,
+  lineHeight,
+  transition,
+  fontSize,
+  duration,
+  easing,
+} from '../../theme'
+import { ChevronDown } from '../../icons'
+import { H3 } from '../Heading'
+
+interface CollapsibleProps {
+  summary: string
+  children: ReactNode
+  defaultOpen?: boolean
+  open?: boolean
+  onToggle?: (open: boolean) => void
+  as?: string | undefined
+}
+
+interface ButtonProps {
+  open: boolean
+}
+interface DetailsProps {
+  open: boolean
+}
+interface PanelWrapperProps {
+  ref: React.MutableRefObject<HTMLDivElement | null>
+  height: string
+}
+
+const Details = styled.div<DetailsProps>`
+  position: relative;
+  text-align: start;
+  width: 100%;
+  background-color: ${color.elevatedBackground};
+  border-radius: ${radius.md};
+`
+
+const Button = styled.button<ButtonProps>`
```

Comment:
Nit: Types need to be above the thing that uses the type

Reply:
Check ✔️